testing/camunda-process-test-java/src/main/java/io/camunda/process/test/api/assertions/ProcessInstanceAssert.java

```diff
@@ -390,6 +392,124 @@ ProcessInstanceAssert hasLocalVariable(
    */
   ProcessInstanceAssert hasVariables(Map<String, Object> variables);
 
+  /**
+   * Verifies that the variable object satisfied the given requirements expressed as a {@link
+   * ThrowingConsumer}.
+   *
+   * <p>This is useful to perform a group of assertions on a single object, each passed assertion is
+   * evaluated and all failures are reported (to be precise each assertion can lead to one failure
+   * max).
+   *
+   * @param variableName the variable name
+   * @param jsonMappedClass the variable's deserialization target, can be a JsonNode or Map for
+   *     variables without a dedicated class
+   * @param requirement the requirement that the variable must satisfy
+   * @return the assertion object
+   */
+  <T> ProcessInstanceAssert hasVariableSatisfies(
+      String variableName, final Class<T> jsonMappedClass, final ThrowingConsumer<T> requirement);
+
+  /**
+   * Verifies that the process instance has a variable with a value that satisfies the given
+   * requirements. The verification fails if the variable doesn't exist or the value doesn't satisfy
+   * all requirements.
+   *
+   * <p>The assertion waits until the variable exists and the value satisfies all requirements.
+   *
+   * @param variableName the variable name
+   * @param jsonMappedClass the variable's deserialization target, can be a JsonNode or Map for
+   *     variables without a dedicated class
+   * @param requirements the list of requirements that the variable must satisfy
+   * @return the assertion object
+   */
+  @SuppressWarnings("unchecked")
+  <T> ProcessInstanceAssert hasVariableSatisfies(
+      String variableName, Class<T> jsonMappedClass, List<ThrowingConsumer<T>> requirements);
```

Comment:
:thought_balloon: We should favor a varargs instead of a list. Passing a list is cumbersome. 

The [assertJ API](https://www.javadoc.io/static/org.assertj/assertj-core/4.0.0-M1/org.assertj.core/org/assertj/core/api/AbstractAssert.html#satisfies(java.util.function.Consumer...)) also uses varargs for the requirements.